enhancements/ingress/configurable-dns-loadbalancerservice.md

```diff
@@ -0,0 +1,199 @@
+provider reports a degraded state because the provided DNS can't be created.
+
+This enhancement aims to provide the end-user the ability to completely disable
+DNS management on an Ingress Controller.
+
+## Motivation
+
+End users need to have an option to manage DNS on their own instead of relying
+on the `cluster-ingress-operator`. This is quite a common requirement where
+1. Customers may need deploy ingress controllers on domains hosted outside of
+   cloud providers.
+2. Partners may need to deploy ingress controllers for domains from their own 
+   customers (with domains not hosted in the same cloud provider).
+
+With no support for such scenarious currently, customers end up with clusters
+in degraded state. The solution to this is to add support
+in the `cluster-ingress-operator` to be able to optionally manage DNS record
+creation. 
+
+### User Stories
+
+#### As a cluster admin, when configuring an ingresscontroller, I want to specify whether a DNS record should or should not be created for this ingress 
+
+The cluster admin has the option to modify/create ingresscontroller to 
+specify whether a DNS record should or should not be created. This is 
+available as a Day-2 operation to the cluster admin.
+
+#### As a cluster admin, when updating an ingresscontroller from managed DNS to unmanaged DNS, I want the previous DNS records to be deleted
+
+Upon updating `manageDNS` to `false` the older DNS records that were created
+must be deleted by the ingress operator.
+
+### Goals
+
+- Provide the ability to opt in for DNS management by the `cluster-ingress-operator`.
+- Recover from degraded state of the `cluster-ingress-operator` during upgrades
+  from OCP 3.x to 4.x where DNS was being managed externally,
+
+### Non-Goals
+
+- Provide a day-0 solution for cluster installations involving external DNS
+  managaments by the customer.
+
+## Proposal
+
+Currently there is partial support provided by the `installer` (in UPI installations)
+to disable the `cluster-ingress-operator` from managing the DNS cluster-wide as documented
+[here](https://github.com/openshift/installer/blob/master/docs/user/aws/install_upi.md#remove-dns-zones). 
+
+The proposed solution adds support for more fine grained control over specific
+ingresscontrollers on how they manage the wildcard DNS records associated with them.
+
+### API Extensions
+
+The IngressController CRD is extended by adding an optional `ManageDNS`
+field with type `bool`. This field will default to true.
+
+```go
+// LoadBalancerStrategy holds parameters for a load balancer.
+type LoadBalancerStrategy struct {
+    // <snip>
+
+    // manageDNS indicates if the lifecyle of the wildcard DNS record
+    // associated with the load balancer service will be managed by
+    // the ingress operator.
+    //
+	// +kubebuilder:default:=true
+	// +kubebuilder:validation:Optional 
+    // +optional
+    ManageDNS bool `json:"manageDNS"`
```

Comment:
Just to foresee the potential remarks from the API team.
Maybe a generalization of the boolean would be better percieved, see also [the Kube conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#primitive-types) for `bool`.

I can think of something like:
```go
// +kubebuilder:default:="managed"
// +kubebuilder:validation:Optional
// +optional
DNS LoadBalancerDNSmanagement `json:"dns"`
}

type LoadBalancerDNSManagement string

var (
    ManagedLoadBalancerDNS LoadBalancerDNSManagement = "managed"
    UnmanagedLoadBalancerDNS LoadBalancerDNSManagement = "unmanaged"
)
```

WDYT?

Reply:
Yeah, this would make sense. Future use case could involve denoting managed by externalDNS. I'll make the required changes.